hashring/hashring.go

```diff
@@ -66,53 +69,46 @@ func (r *HashRing) RegisterListener(l events.EventListener) {
 	r.listeners = append(r.listeners, l)
 }
 
-func NewHashRing(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
+// New instantiates and returns a new HashRing,
+func New(hashfunc func([]byte) uint32, replicaPoints int) *HashRing {
 	ring := &HashRing{
-		hashfunc:      hashfunc,
 		replicaPoints: replicaPoints,
+		hashfunc: func(str string) int {
+			return int(hashfunc([]byte(str)))
+		},
 	}
 
 	ring.servers.byAddress = make(map[string]bool)
-	ring.servers.tree = &RBTree{}
-
+	ring.servers.tree = new(RedBlackTree)
```

Comment:
Any reason for switching to `new`? I've been using the &Foo{} syntax because that is what is used elsewhere in this codebase.